### ISSUE TITLE
FinalLocalVariable: "throw" false negative

### ISSUE THREAD

**Issue by Luolc** [checkstyle/checkstyle#4225]
Taken from https://github.com/checkstyle/checkstyle/pull/4115#discussion_r110134567

```
$ javac Foo.java
$ cat Foo.java
public class Foo {
    void foo() {
        int a; // expecte a violation
        final int random = (int) Math.random();
        if (random > 0) {
            a = 1;
            throw new RuntimeException();
        }
        a = 2;
    }
}
```

```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="FinalLocalVariable">
        </module>
    </module>
</module>
```

```
$ java -jar checkstyle-7.7-SNAPSHOT-all.jar -c config.xml Foo.java
Starting audit...
Audit done.
```

Expect: a violation at line 3.

**Comment by DXTkastb** [comment]
I'm on it.
